apps/pipelines/models.py

```diff
@@ -418,19 +426,28 @@ def create_new_version(self):
         and update the `assistant_id` in the node params to the new assistant version id.
         """
         from apps.assistants.models import OpenAiAssistant
-        from apps.pipelines.nodes.nodes import AssistantNode
-
-        assistant_node_name = AssistantNode.__name__
+        from apps.documents.models import Collection
+        from apps.pipelines.nodes.nodes import AssistantNode, LLMResponseWithPrompt
 
         new_version = super().create_new_version(save=False)
 
-        if self.type == assistant_node_name and new_version.params.get("assistant_id"):
+        if self.type == AssistantNode.__name__ and new_version.params.get("assistant_id"):
             assistant = OpenAiAssistant.objects.get(id=new_version.params.get("assistant_id"))
             if not assistant.is_a_version:
                 assistant_version = assistant.create_new_version()
                 # convert to string to be consistent with values from the UI
                 new_version.params["assistant_id"] = str(assistant_version.id)
 
+        if self.type == LLMResponseWithPrompt.__name__:
+            if collection_id := self.params.get("collection_id"):
+                collection = Collection.objects.get(id=collection_id)
+
+                if collection.has_versions is False or collection.compare_with_latest():
```

Comment:
whats the runtime of collection.compare_with_latest() with large files? I assume .compare() would take a significant amount of time?

Reply:
Good question. Since the file model also implements `version_details`, we're using that when comparing, and since version details consists only of of the name and the summary of each file, it should have a small memory footprint. The only reason why one file might be a version of another is when the name or summary of the file changed.